game-core/src/test/java/org/triplea/client/ui/javafx/TripleAApplicationTest.java

```diff
@@ -0,0 +1,43 @@
+package org.triplea.client.ui.javafx;
+
+import static org.testfx.api.FxAssert.verifyThat;
+import static org.testfx.matcher.base.NodeMatchers.hasChildren;
+import static org.testfx.matcher.base.NodeMatchers.isInvisible;
+import static org.testfx.matcher.base.NodeMatchers.isVisible;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.sonatype.goodies.prefs.memory.MemoryPreferences;
+import org.testfx.api.FxToolkit;
+import org.testfx.framework.junit5.ApplicationExtension;
+
+import games.strategy.triplea.settings.ClientSetting;
+
+@ExtendWith(ApplicationExtension.class)
+public class TripleAApplicationTest {
+
+  static {
+    System.setProperty("testfx.robot", "glass");
+    System.setProperty("testfx.headless", String.valueOf(true));
+    System.setProperty("prism.order", "sw");
+    System.setProperty("prism.text", "t2k");
+    System.setProperty("testfx.setup.timeout", String.valueOf(10000));
+  }
+
+
+  @BeforeAll
+  public static void setup() throws Exception {
+    ClientSetting.setPreferences(new MemoryPreferences());
```

Comment:
Note that you need to call `ClientSetting#resetPreferences()` in a `@AfterAll` method, otherwise it may interfere with other `ClientSetting` usage in other tests depending on execution order.

However, instead of calling `ClientSetting#setPreferences()` directly, I'd recommend creating a new class similar to `AbstractClientSettingTestCase`, except that it initializes the preferences in a `@BeforeAll` method and uninitializes them in a `@AfterAll` method.  That way you can keep all the `MemoryPreferences` stuff encapsulated in that package.  If you write many JFX test fixtures, you'll probably need similar functionality in all those classes anyway.

This worked for me:

```diff
diff --git a/game-core/src/test/java/games/strategy/triplea/settings/AbstractClientSettingTestCase.java b/game-core/src/test/java/games/strategy/triplea/settings/AbstractClientSettingTestCase.java
index ffc9886a2..9cd81ed7b 100644
--- a/game-core/src/test/java/games/strategy/triplea/settings/AbstractClientSettingTestCase.java
+++ b/game-core/src/test/java/games/strategy/triplea/settings/AbstractClientSettingTestCase.java
@@ -5,7 +5,8 @@ import org.junit.jupiter.api.BeforeEach;
 import org.sonatype.goodies.prefs.memory.MemoryPreferences;
 
 /**
- * Superclass for test fixtures that directly or indirectly interact with instances of {@link ClientSetting}.
+ * Superclass for test fixtures that directly or indirectly interact with instances of {@link ClientSetting} and require
+ * the preference store to be reset between test runs.
  *
  * <p>
  * This fixture ensures the {@link ClientSetting} preferences are properly initialized before each test and
diff --git a/game-core/src/test/java/games/strategy/triplea/settings/AbstractPerFixtureClientSettingTestCase.java b/game-core/src/test/java/games/strategy/triplea/settings/AbstractPerFixtureClientSettingTestCase.java
new file mode 100644
index 000000000..4e42a6682
--- /dev/null
+++ b/game-core/src/test/java/games/strategy/triplea/settings/AbstractPerFixtureClientSettingTestCase.java
@@ -0,0 +1,28 @@
+package games.strategy.triplea.settings;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.sonatype.goodies.prefs.memory.MemoryPreferences;
+
+/**
+ * Superclass for test fixtures that directly or indirectly interact with instances of {@link ClientSetting} and require
+ * the preference store to persist between test runs.
+ *
+ * <p>
+ * This fixture ensures the {@link ClientSetting} preferences are properly initialized before all tests are run and
+ * uninitialized after all tests are complete.
+ * </p>
+ */
+public abstract class AbstractPerFixtureClientSettingTestCase {
+  protected AbstractPerFixtureClientSettingTestCase() {}
+
+  @BeforeAll
+  public static void initializeClientSettingPreferences() {
+    ClientSetting.setPreferences(new MemoryPreferences());
+  }
+
+  @AfterAll
+  public static void uninitializeClientSettingPreferences() {
+    ClientSetting.resetPreferences();
+  }
+}
diff --git a/game-core/src/test/java/org/triplea/client/ui/javafx/TripleAApplicationTest.java b/game-core/src/test/java/org/triplea/client/ui/javafx/TripleAApplicationTest.java
index e4164a03e..c91af3106 100644
--- a/game-core/src/test/java/org/triplea/client/ui/javafx/TripleAApplicationTest.java
+++ b/game-core/src/test/java/org/triplea/client/ui/javafx/TripleAApplicationTest.java
@@ -8,14 +8,13 @@ import static org.testfx.matcher.base.NodeMatchers.isVisible;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
-import org.sonatype.goodies.prefs.memory.MemoryPreferences;
 import org.testfx.api.FxToolkit;
 import org.testfx.framework.junit5.ApplicationExtension;
 
-import games.strategy.triplea.settings.ClientSetting;
+import games.strategy.triplea.settings.AbstractPerFixtureClientSettingTestCase;
 
 @ExtendWith(ApplicationExtension.class)
-public class TripleAApplicationTest {
+public class TripleAApplicationTest extends AbstractPerFixtureClientSettingTestCase {
 
   static {
     System.setProperty("testfx.robot", "glass");
@@ -28,7 +27,6 @@ public class TripleAApplicationTest {
 
   @BeforeAll
   public static void setup() throws Exception {
-    ClientSetting.setPreferences(new MemoryPreferences());
     FxToolkit.registerPrimaryStage();
     FxToolkit.setupApplication(TripleA.class);
   }
```

Not shown above is that I would have renamed `AbstractClientSettingTestCase` to `AbstractPerTestClientSettingTestCase` to balance the new class name.  There's probably better adjectives for those two superclasses, so feel free to rename them if you can think of something more descriptive.

Reply:
Another thing I noticed is that if you run the tests without Monocle (so that the JFX app is actually rendered on-screen), the app stays visible until the entire `junitPlatformTest` task is complete.  Since you're calling `FxToolkit#setupApplication()` in a `@BeforeAll`, is there a corresponding method you can call in a `@AfterAll` to ensure the application is terminated when this test fixture is finished?